src/sinks/util/service.rs

```diff
@@ -63,18 +63,40 @@ impl<L> ServiceBuilderExt<L> for ServiceBuilder<L> {
     }
 }
 
-/// Tower Request based configuration
-#[derive(Clone, Copy, Debug, Deserialize, Serialize)]
+/// Middleware settings for outbound requests.
+///
+/// Various settings can be configured, such as concurrency and rate limits, timeouts, etc.
+#[configurable_component]
+#[derive(Clone, Copy, Debug)]
 pub struct TowerRequestConfig {
+    #[configurable(derived)]
     #[serde(default)]
     #[serde(skip_serializing_if = "concurrency_is_none")]
-    pub concurrency: Concurrency, // adaptive
-    pub timeout_secs: Option<u64>,             // 1 minute
-    pub rate_limit_duration_secs: Option<u64>, // 1 second
-    pub rate_limit_num: Option<u64>,           // i64::MAX
-    pub retry_attempts: Option<usize>,         // isize::MAX
+    pub concurrency: Concurrency,
+
+    /// The maximum time a request can take before being aborted. It is highly recommended that you do not lower this value below the service’s internal timeout, as this could create orphaned requests, pile on retries, and result in duplicate data downstream.
+    pub timeout_secs: Option<u64>,
```

Comment:
Nit: Suggesting to add a couple of linebreaks here. Or does it break the `configurable_component` comment somehow?

Reply:
Looks like I just missed a spot. Normally I would be adding line breaks to generate the intended title vs description separation.

Reply:
Fixed in 2d2b23f55.